File: backend/internal/model/resume.go
package model

import "encoding/json"

// ResumeListItem 简历列表项
type ResumeListItem struct {
	ID        string `json:"id"`
	Title     string `json:"title"`
	Template  string `json:"template"`
	UpdatedAt int64  `json:"updatedAt"`
	CreatedAt int64  `json:"createdAt"`
}

// ResumeContent 简历内容结构
type ResumeContent struct {
	ThemeColor        string                   `json:"themeColor"`
	StyleSettings     ResumeStyleSettings      `json:"styleSettings"`
	Modules           []map[string]interface{} `json:"modules"`
	BasedOnSnapshotID *string                  `json:"basedOnSnapshotId,omitempty"`
}

// ResumeStyleSettings 简历样式设置
type ResumeStyleSettings struct {
	FontFamily               string  `json:"fontFamily"`
	FontSize                 float64 `json:"fontSize"`
	TextColor                string  `json:"textColor"`
	LineHeight               float64 `json:"lineHeight"`
	PagePaddingHorizontal    float64 `json:"pagePaddingHorizontal"`
	PagePaddingVertical      float64 `json:"pagePaddingVertical"`
	ModuleSpacing            float64 `json:"moduleSpacing"`
	ParagraphSpacing         float64 `json:"paragraphSpacing"`
	ModuleTitleLinePosition  string  `json:"moduleTitleLinePosition"`
	ModuleTitleMarkerStyle   string  `json:"moduleTitleMarkerStyle"`
	ModuleTitleMarkerVisible bool    `json:"moduleTitleMarkerVisible"`
}

// ResumeDetail 简历详情
type ResumeDetail struct {
	ID                string                     `json:"id"`
	Title             string                     `json:"title"`
	Locale            string                     `json:"locale"`
	Template          string                     `json:"template"`
	ThemeColor        string                     `json:"themeColor"`
	StyleSettings     ResumeStyleSettings        `json:"styleSettings"`
	Modules           []map[string]interface{}   `json:"modules"`
	LatestVersionID   *string                    `json:"latestVersionId"`
	LatestSnapshotID  *string                    `json:"latestSnapshotId,omitempty"`
	BasedOnSnapshotID *string                    `json:"basedOnSnapshotId,omitempty"`
	SnapshotDrafts    map[string]json.RawMessage `json:"snapshotDrafts,omitempty"`
	UpdatedAt         int64                      `json:"updatedAt"`
	CreatedAt         int64                      `json:"createdAt"`
}

// CreateResumeRequest 创建简历请求
type CreateResumeRequest struct {
	Title         string                   `json:"title" binding:"required,max=255"`
	Locale        string                   `json:"locale" binding:"max=10"`
	Template      string                   `json:"template" binding:"max=50"`
	ThemeColor    string                   `json:"themeColor"`
	StyleSettings *ResumeStyleSettings     `json:"styleSettings"`
	Modules       []map[string]interface{} `json:"modules"`
}

// UpdateResumeRequest 更新简历请求
type UpdateResumeRequest struct {
	Title             string                     `json:"title" binding:"max=255"`
	ThemeColor        string                     `json:"themeColor"`
	StyleSettings     *ResumeStyleSettings       `json:"styleSettings"`
	Modules           []map[string]interface{}   `json:"modules"`
	ClientUpdatedAt   int64                      `json:"clientUpdatedAt"`
	BasedOnSnapshotID *string                    `json:"basedOnSnapshotId,omitempty"`
	SnapshotDrafts    map[string]json.RawMessage `json:"snapshotDrafts,omitempty"`
}

// Pagination 分页信息
type Pagination struct {
	Page       int `json:"page"`
	PageSize   int `json:"pageSize"`
	Total      int `json:"total"`
	TotalPages int `json:"totalPages"`
}

// ResumeListResponse 简历列表响应
type ResumeListResponse struct {
	Items      []ResumeListItem `json:"items"`
	Pagination Pagination       `json:"pagination"`
}

// ResumeUpdateResponse 更新简历响应
type ResumeUpdateResponse struct {
	ID               string  `json:"id"`
	UpdatedAt        int64   `json:"updatedAt"`
	LatestVersionID  *string `json:"latestVersionId"`
	LatestSnapshotID *string `json:"latestSnapshotId,omitempty"`
}

// ---------- 版本快照相关类型 ----------

// SnapshotType 快照类型
type SnapshotType string

const (
	SnapshotTypeAuto    SnapshotType = "auto"
	SnapshotTypeManual  SnapshotType = "manual"
	SnapshotTypeDefault SnapshotType = "default"
)

// VersionSnapshot 版本快照
type VersionSnapshot struct {
	ID           string       `json:"id"`
	ResumeID     string       `json:"resumeId"`
	UserID       string       `json:"userId"`
	VersionNo    int          `json:"versionNo"`
	SnapshotType SnapshotType `json:"snapshotType"`
	Label        *string      `json:"label,omitempty"`
	JDContextID  *string      `json:"jdContextId,omitempty"`
	CreatedAt    int64        `json:"createdAt"`
	IsCurrent    bool         `json:"isCurrent"`
}

// SnapshotListItem 快照列表项
type SnapshotListItem struct {
	ID           string       `json:"id"`
	VersionNo    int          `json:"versionNo"`
	SnapshotType SnapshotType `json:"snapshotType"`
	Label        *string      `json:"label,omitempty"`
	CreatedAt    int64        `json:"createdAt"`
	IsCurrent    bool         `json:"isCurrent"`
}

// CreateSnapshotRequest 创建手动快照
type CreateSnapshotRequest struct {
	Label string `json:"label" binding:"required,max=100"`
}

// UpdateSnapshotRequest 更新快照标签
type UpdateSnapshotRequest struct {
	Label string `json:"label" binding:"required,max=100"`
}

// SnapshotListResponse 快照列表响应
type SnapshotListResponse struct {
	Items   []SnapshotListItem `json:"items"`
	Total   int                `json:"total"`
	HasMore bool               `json:"hasMore"`
}

// DiffResult 快照对比结果
type DiffResult struct {
	SnapshotA SnapshotBrief `json:"snapshotA"`
	SnapshotB SnapshotBrief `json:"snapshotB"`
	Diffs     []FieldDiff   `json:"diffs"`
	Stats     DiffStats     `json:"stats"`
}

// SnapshotBrief 快照简要信息
type SnapshotBrief struct {
	ID        string  `json:"id"`
	VersionNo int     `json:"versionNo"`
	Label     *string `json:"label,omitempty"`
	CreatedAt int64   `json:"createdAt"`
}

// FieldDiff 字段级差异
type FieldDiff struct {
	ModuleType       string `json:"moduleType"`
	ModuleInstanceID string `json:"moduleInstanceId"`
	Field            string `json:"field"`
	Before           string `json:"before"`
	After            string `json:"after"`
}

// DiffStats 对比统计
type DiffStats struct {
	ModulesAdded    int `json:"modulesAdded"`
	ModulesRemoved  int `json:"modulesRemoved"`
	ModulesModified int `json:"modulesModified"`
	FieldsChanged   int `json:"fieldsChanged"`
}

// DiffSnapshotsRequest 对比请求
type DiffSnapshotsRequest struct {
	SnapshotAID       string                   `json:"snapshotAId" binding:"required"`
	SnapshotBID       string                   `json:"snapshotBId" binding:"required"`
	CurrentModules    []map[string]interface{} `json:"currentModules,omitempty"`
	ComparisonModules []map[string]interface{} `json:"comparisonModules,omitempty"`
}

// ============ 分享链接 ============

// ShareLink 分享链接
type ShareLink struct {
	ID         string  `json:"id"`
	ResumeID   string  `json:"resumeId"`
	Token      string  `json:"token"`
	CreatedBy  string  `json:"createdBy"`
	SnapshotID *string `json:"snapshotId,omitempty"`
	ExpiresAt  *int64  `json:"expiresAt,omitempty"`
	ViewCount  int     `json:"viewCount"`
	IsActive   bool    `json:"isActive"`
	CreatedAt  int64   `json:"createdAt"`
	ShareURL   string  `json:"shareUrl,omitempty"`
}

// ShareComment 分享评论
type ShareComment struct {
	ID         string  `json:"id"`
	ShareID    string  `json:"shareId"`
	SnapshotID *string `json:"snapshotId,omitempty"`
	VisitorID  string  `json:"-"`
	AuthorName string  `json:"authorName"`
	Content    string  `json:"content"`
	ModuleID   string  `json:"moduleId"`
	ItemIndex  int     `json:"itemIndex"`
	CreatedAt  int64   `json:"createdAt"`
}

// CreateShareRequest 创建分享请求
type CreateShareRequest struct {
	ResumeID   string `json:"resumeId" binding:"required"`
	SnapshotID string `json:"snapshotId,omitempty"`
	ExpiresIn  int    `json:"expiresIn"`
}

// AddCommentRequest 添加评论请求
type AddCommentRequest struct {
	AuthorName string  `json:"authorName"`
	Content    string  `json:"content" binding:"required"`
	ModuleID   string  `json:"moduleId"`
	ItemIndex  int     `json:"itemIndex"`
	VisitorID  string  `json:"visitorId"`
	SnapshotID *string `json:"snapshotId,omitempty"`
}

// ShareResumeView 分享页简历数据
type ShareResumeView struct {
	Title            string                   `json:"title"`
	Locale           string                   `json:"locale"`
	ThemeColor       string                   `json:"themeColor"`
	Modules          []map[string]interface{} `json:"modules"`
	LatestSnapshotID *string                  `json:"latestSnapshotId,omitempty"`
	ShareInfo        *ShareLink               `json:"shareInfo"`
	Comments         []ShareComment           `json:"comments"`
}

// AIAnalysisRequest AI 分析请求
type AIAnalysisRequest struct {
	Token string `json:"token" binding:"required"`
}

// AIAnalysisResponse AI 分析响应
type AIAnalysisResponse struct {
	Summary     string   `json:"summary"`
	Strengths   []string `json:"strengths"`
	Weaknesses  []string `json:"weaknesses"`
	Suggestions []string `json:"suggestions"`
}

// RequirementDocRequest 需求文档生成请求
type RequirementDocRequest struct {
	Token string `json:"token" binding:"required"`
}

// ============ 管理员评论视图 ============

// AdminCommentItem 管理员视角的单条评论
type AdminCommentItem struct {
	ID            string  `json:"id"`
	ShareToken    string  `json:"shareToken"`
	VisitorID     string  `json:"visitorId"`
	AuthorName    string  `json:"authorName"`
	Content       string  `json:"content"`
	ModuleID      string  `json:"moduleId"`
	ModuleTitle   string  `json:"moduleTitle"`
	ItemIndex     int     `json:"itemIndex"`
	ItemLabel     string  `json:"itemLabel"`
	SnapshotID    *string `json:"snapshotId,omitempty"`
	SnapshotLabel *string `json:"snapshotLabel,omitempty"`
	CreatedAt     int64   `json:"createdAt"`
}

// ModuleCommentSummary 模块评论汇总
type ModuleCommentSummary struct {
	ModuleID     string `json:"moduleId"`
	ModuleTitle  string `json:"moduleTitle"`
	CommentCount int    `json:"commentCount"`
	VisitorCount int    `json:"visitorCount"`
}

// AdminCommentsSummary 统计信息
type AdminCommentsSummary struct {
	TotalComments   int                    `json:"totalComments"`
	TotalVisitors   int                    `json:"totalVisitors"`
	ModuleBreakdown []ModuleCommentSummary `json:"moduleBreakdown"`
}

// AdminCommentsResponse 管理员评论总览响应
type AdminCommentsResponse struct {
	Items   []AdminCommentItem   `json:"items"`
	Summary AdminCommentsSummary `json:"summary"`
}

File: src/api/resume.ts
// ============================================================
// 简历 API
// ============================================================

import { apiClient } from './client'
import type {
  ResumeListResponse,
  ResumeDetail,
  CreateResumeRequest,
  UpdateResumeRequest,
  ResumeUpdateResponse,
} from './types'

export interface ListResumesParams {
  page?: number
  pageSize?: number
  keyword?: string
}

export interface AdminCommentItem {
  id: string
  shareToken: string
  visitorId: string
  authorName: string
  content: string
  moduleId: string
  moduleTitle: string
  itemIndex: number
  itemLabel: string
  createdAt: number
}

export interface ModuleCommentSummary {
  moduleId: string
  moduleTitle: string
  commentCount: number
  visitorCount: number
}

export interface AdminCommentsResponse {
  items: AdminCommentItem[]
  summary: {
    totalComments: number
    totalVisitors: number
    moduleBreakdown: ModuleCommentSummary[]
  }
}

export const resumeApi = {
  list: (params?: ListResumesParams) => {
    const searchParams = new URLSearchParams()
    if (params?.page) searchParams.set('page', String(params.page))
    if (params?.pageSize) searchParams.set('pageSize', String(params.pageSize))
    if (params?.keyword) searchParams.set('keyword', params.keyword)
    const query = searchParams.toString()
    return apiClient.get<ResumeListResponse>(
      `/resumes${query ? `?${query}` : ''}`
    )
  },

  create: (data: CreateResumeRequest) =>
    apiClient.post<ResumeDetail>('/resumes', data),

  get: (id: string) => apiClient.get<ResumeDetail>(`/resumes/${id}`),

  update: (id: string, data: UpdateResumeRequest) =>
    apiClient.put<ResumeUpdateResponse>(`/resumes/${id}`, data),

  delete: (id: string) =>
    apiClient.delete<{ deleted: boolean }>(`/resumes/${id}`),

  // ---------- 版本快照 ----------

  getSnapshots: (
    id: string,
    params?: { limit?: number; includeAuto?: boolean }
  ) => {
    const searchParams = new URLSearchParams()
    if (params?.limit) searchParams.set('limit', String(params.limit))
    if (params?.includeAuto !== undefined)
      searchParams.set('includeAuto', String(params.includeAuto))
    const query = searchParams.toString()
    return apiClient.get<
      { items: SnapshotListItem[]; total: number; hasMore: boolean }
    >(`/resumes/${id}/snapshots${query ? `?${query}` : ''}`)
  },

  createSnapshot: (resumeId: string, label: string) =>
    apiClient.post<SnapshotDetail>(`/resumes/${resumeId}/snapshots`, { label }),

  updateSnapshotLabel: (resumeId: string, snapshotId: string, label: string) =>
    apiClient.put<{ updated: boolean }>(
      `/resumes/${resumeId}/snapshots/${snapshotId}`,
      { label }
    ),

  deleteSnapshot: (resumeId: string, snapshotId: string) =>
    apiClient.delete<{ deleted: boolean }>(
      `/resumes/${resumeId}/snapshots/${snapshotId}`
    ),

  getSnapshotDetail: (resumeId: string, snapshotId: string) =>
    apiClient.get<{ snapshot: SnapshotDetail; content: unknown }>(
      `/resumes/${resumeId}/snapshots/${snapshotId}`
    ),

  restoreFromSnapshot: (resumeId: string, snapshotId: string) =>
    apiClient.post<ResumeUpdateResponse>(
      `/resumes/${resumeId}/snapshots/${snapshotId}/restore`
    ),

  diffSnapshots: (resumeId: string, aId: string, bId: string, currentModules?: unknown[], comparisonModules?: unknown[]) =>
    apiClient.post<DiffResult>(`/resumes/${resumeId}/snapshots/diff`, {
      snapshotAId: aId,
      snapshotBId: bId,
      currentModules: currentModules || undefined,
      comparisonModules: comparisonModules || undefined,
    }),

  // 评论管理（管理员视图）
  getComments: (resumeId: string) =>
    apiClient.get<AdminCommentsResponse>(`/resumes/${resumeId}/comments`, { auth: true }),
}

// ---------- 快照相关类型 ----------

export type SnapshotType = 'auto' | 'manual' | 'default'

export interface SnapshotListItem {
  id: string
  versionNo: number
  snapshotType: SnapshotType
  label?: string
  createdAt: number
  isCurrent: boolean
}

export interface SnapshotDetail {
  id: string
  resumeId: string
  versionNo: number
  snapshotType: SnapshotType
  label?: string
  createdAt: number
  isCurrent: boolean
}

export interface DiffResult {
  snapshotA: { id: string; versionNo: number; label?: string; createdAt: number }
  snapshotB: { id: string; versionNo: number; label?: string; createdAt: number }
  diffs: FieldDiff[]
  stats: DiffStats
}

export interface FieldDiff {
  moduleType: string
  moduleInstanceId: string
  field: string
  before: string
  after: string
}

export interface DiffStats {
  modulesAdded: number
  modulesRemoved: number
  modulesModified: number
  fieldsChanged: number
}

// ============ 分享链接 ============

export interface ShareLink {
  id: string
  resumeId: string
  token: string
  createdBy: string
  expiresAt?: number
  viewCount: number
  isActive: boolean
  createdAt: number
  shareUrl?: string
}

export interface ShareComment {
  id: string
  shareId: string
  authorName: string
  content: string
  moduleId: string
  itemIndex: number
  createdAt: number
}

export interface ShareResumeView {
  title: string
  locale: string
  themeColor: string
  modules: unknown[]
  shareInfo: ShareLink
  comments: ShareComment[]
}

export interface AIAnalysisResponse {
  summary: string
  strengths: string[]
  weaknesses: string[]
  suggestions: string[]
}

export const shareApi = {
  create: (resumeId: string, expiresIn?: number) =>
    apiClient.post<ShareLink>(`/resumes/${resumeId}/share`, { resumeId, expiresIn: expiresIn ?? 0 }, { auth: true }),

  list: (resumeId: string) =>
    apiClient.get<{ items: ShareLink[] }>(`/resumes/${resumeId}/shares`, { auth: true }),

  deactivate: (resumeId: string, shareId: string) =>
    apiClient.delete<{ deactivated: boolean }>(`/resumes/${resumeId}/shares/${shareId}`, { auth: true }),

  view: (token: string) =>
    apiClient.get<ShareResumeView>(`/share/${token}`),

  addComment: (token: string, content: string, authorName?: string, moduleId?: string, itemIndex?: number, visitorId?: string) =>
    apiClient.post<ShareComment>(`/share/${token}/comments`, { content, authorName: authorName || '', moduleId: moduleId || '', itemIndex: itemIndex ?? 0, visitorId: visitorId || '' }),

  listComments: (token: string, visitorId?: string) =>
    apiClient.get<{ items: ShareComment[] }>(`/share/${token}/comments${visitorId ? `?visitorId=${encodeURIComponent(visitorId)}` : ''}`),

  analyze: (token: string) =>
    apiClient.post<AIAnalysisResponse>(`/share/${token}/analyze`),

  generateRequirementDoc: (token: string) =>
    apiClient.post<{ document: string }>(`/share/${token}/requirement-doc`),

  downloadPDF: async (token: string, html: string, filename: string) => {
    const res = await fetch(`/api/share/${token}/pdf`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ html, filename }),
    })
    if (!res.ok) throw new Error('PDF download failed')
    const blob = await res.blob()
    const url = URL.createObjectURL(blob)
    const a = document.createElement('a')
    a.href = url
    a.download = `${filename}.pdf`
    a.click()
    URL.revokeObjectURL(url)
  },
}